modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C

```diff
@@ -928,5 +948,112 @@ PolygonConcentricCircleMeshGeneratorBase::generate()
   if (flat_side_up)
     MeshTools::Modification::rotate(*mesh0, 180.0 / (Real)_num_sides, 0.0, 0.0);
   mesh0->set_isnt_prepared();
+
+  if (_ring_block_ids.size() > 2 && getParam<bool>("replace_inner_ring_with_delaunay_mesh"))
+  {
+    // remove elements of the inner ring and add the boundary side set
+    std::set<Elem *> deleteable_elems;
+    for (auto & elem : mesh0->element_ptr_range())
+      if (elem->vertex_average().norm() < _ring_radii[0])
+        deleteable_elems.insert(elem);
+
+    const boundary_id_type boundary_id = MooseMeshUtils::getBoundaryIDs(*mesh0, {"_foo"}, true)[0];
+    BoundaryInfo & boundary_info = mesh0->get_boundary_info();
+    for (auto & elem : deleteable_elems)
+    {
+      unsigned int n_sides = elem->n_sides();
+      for (unsigned int n = 0; n != n_sides; ++n)
+      {
+        Elem * neighbor = elem->neighbor_ptr(n);
+        if (!neighbor)
```

Comment:
why would this happen? should there be an assert if we dont expect it it to happen?

Reply:
Yes, this looks should not happen unless the pin is cut before reaching this point.

Reply:
eh, I am not so sure. I think this can happen during the process of removing element one by one.